### PR TITLE
refactor(ios): error definitions, i18n

### PR DESCRIPTION
**Pull request by jahorton**
All Keyman-generated errors now have defined types and/or enums devoted to their maintenance, along with localized messages.  Some errors also accept parameters, though these aren't passed along to the messages.  The parameters would be far more useful to programmers (who can condition on the enums and extract said parameters) than end-users, who want a simple, minimally-confusing localized message instead.

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngine/Classes/Constants.swift [hunk 1]
```diff
@@ -124,3 +124,22 @@ public enum FileExtensions {
...
+
+public enum KeymanError: Error, LocalizedError {
+  case unknown
+  // file that is missing, is the file critical?
+  case missingFile(String, Bool)
+
+  var localizedDescription: String {
+    switch self {
+      case .unknown:
+        return engineBundle.localizedString(forKey: "error-unknown", value: nil, table: nil)
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/Constants.swift:136]: Can the pre-localized error description get sent to Sentry?

Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Constants.swift:136]: Probably should convert a lot of the `log.error` logging into Sentry reports, but that's a separate target for another day, not for this PR.

File: ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift [hunk 2]
```diff
@@ -14,15 +14,18 @@ public class Queries {
...
-          return "no data returned"
-        case .networkError(let baseError):
-          return "network error occurred - \(String(describing: baseError))"
-        case .parsingError(let baseError):
-          return "failure occurred when parsing results - \(String(describing: baseError))"
+          return engineBundle.localizedString(forKey: "error-query-no-data", value: nil, table: nil)
+        case .networkError(_):
+          return engineBundle.localizedString(forKey: "error-query-general", value: nil, table: nil)
+        case .parsingError(_):
+          return engineBundle.localizedString(forKey: "error-query-decoding", value: nil, table: nil)
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift:26]:
```suggestion
          return engineBundle.localizedString(forKey: "error-query-parsing", value: nil, table: nil)
```

File: ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift [hunk 3]
```diff
@@ -14,15 +14,18 @@ public class Queries {
...
       switch self {
         case .noData:
-          return "no data returned"
-        case .networkError(let baseError):
-          return "network error occurred - \(String(describing: baseError))"
-        case .parsingError(let baseError):
-          return "failure occurred when parsing results - \(String(describing: baseError))"
+          return engineBundle.localizedString(forKey: "error-query-no-data", value: nil, table: nil)
+        case .networkError(_):
+          return engineBundle.localizedString(forKey: "error-query-general", value: nil, table: nil)
```
Comment (darcywong00) [inline ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift:24]: Why is the key for network error a "general" one?

Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift:24]: https://github.com/keymanapp/keyman/blob/8b47386ca05a8e7116f1ddf579ee614920845c66/ios/engine/KMEI/KeymanEngine/en.lproj/Localizable.strings#L58-L59

I felt it wiser to go for a more general error for any alert messaging.  This case arises when the system reports network errors, so it's kind of hard for me to anticipate ways to provide meaningful error messages here... at least in ways that wouldn't seem like major technobabble to average users.

For programmers, there's an attached parameter (of type `Error`) that contains extra data that could be used in, say, Sentry reports.

File: ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift (hunk 2 above)
Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift:26]: https://github.com/keymanapp/keyman/blob/8b47386ca05a8e7116f1ddf579ee614920845c66/ios/engine/KMEI/KeymanEngine/en.lproj/Localizable.strings#L55-L56

Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift:26]: This is intentionally the same as the other message, much for the same reasons as in that other comment.  A user isn't going to care whether or not it's our custom parsing code vs Swift's default decoding functionality.  We care about that, but they don't.

**Comment by jahorton** [comment]
> I'd like to make sure that errors (such as missing files) are captured in Sentry. Is that going to happen in a future PR? Apart from that, LGTM.

How high of a priority would we like to make this?  Should I redirect to that now, or focus on my current tasks first, waiting until a future sprint?

**Comment by mcdurdin** [comment]
> How high of a priority would we like to make this? Should I redirect to that now, or focus on my current tasks first, waiting until a future sprint?

Future sprint. Just capture this 👍

**Comment by jahorton** [comment]
Logged as #3474.

**Pull request merged**